pkg/modprovider/server.go

```diff
@@ -504,6 +512,7 @@ func (s *server) Read(
 	ctx context.Context,
 	req *pulumirpc.ReadRequest,
 ) (*pulumirpc.ReadResponse, error) {
+	ctx = tfsandbox.WithLogger(ctx, tfbridge.NewTerraformLogRedirector(ctx, s.hostClient))
```

Comment:
This might make sense to reuse yes, the old TF providers still have the habit of emitting `[WARN]` messages on stdout. This is what the redirector takes care of.  In the bridge itself for newer providers there is no need to parse stdout as they can use a more precise typed API but it is still around. Here though it is even more interesting.